### PR TITLE
Implementation of basic authentication for metrics

### PR DESCRIPTION
**Pull request by andrii-hryniv**
Goals:
- Enables Basic authentication for "/metrics" path when "basic_auth_username" and "basic_auth_password" ENV variables are provided;
- Uses variables above for [scrape_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config)  to set Authorization header

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -152,9 +183,15 @@ func main() {
...
+		http.Handle("/metrics", BasicAuth(promhttp.Handler()))
+	} else {
+		http.Handle("/metrics", promhttp.Handler())
+	}
+
 	// Self monitoring
-	http.Handle("/metrics", promhttp.Handler())
-	_ = http.ListenAndServe(fmt.Sprintf(":%d", listenPort), nil)
+	err = http.ListenAndServe(fmt.Sprintf(":%d", listenPort), nil)
+	fmt.Printf("%s", err.Error())
```
Comment (loafoe) [inline main.go:194]: Make this printout conditional i.e. only if err != nil

File: main.go [hunk 2]
```diff
@@ -60,6 +62,33 @@ func (m metrics) IncErrorIncursions() {
 	m.ErrorIncursions.Inc()
 }
 
+func MetricsEndpointBasicAuthEnabled() bool {
```
Comment (loafoe) [inline main.go:65]: This function is introduced twice, use the one in the `utils` package

**Review by loafoe — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (andrii-hryniv) [reply main.go]: Removed duplicated code

File: main.go (hunk 1 above)
Comment (andrii-hryniv) [reply main.go]: Added if statement

**Review by loafoe — APPROVED** [review]

**Pull request merged**